e2e2/test/cases/neuron/manifests/bert-train.yaml

```diff
@@ -0,0 +1,65 @@
+apiVersion: kubeflow.org/v2beta1
+kind: MPIJob
+metadata:
+  name: bert-mpi-training
+spec:
+  slotsPerWorker: 1
+  runPolicy:
+    backoffLimit: 20
+    cleanPodPolicy: Running
+  mpiReplicaSpecs:
+    Launcher:
+      replicas: 1
+      template:
+        spec:
+          restartPolicy: OnFailure
+          containers:
+          - image: public.ecr.aws/o5d5x8n6/weicongw:neuron
+            imagePullPolicy: Always
+            name: bert-training-launcher
+            env:
+            - name: NEURON_CORES
+              value: "32"
+            command:
+            - /opt/amazon/openmpi/bin/mpirun
+            - --allow-run-as-root
+            - -np
+            - "{{.WorkerNodeCount}}"
```

Comment:
Can you provide what value was used for your test here? (i.e. "{{.WorkerNodeCount}}")

I don't see it in the command you posted. The number of process for your node count in the test (i.e. 32 processes for 2 nodes w/ 16 cores each) is correct. But `"{{.WorkerNodeCount}}"` to me sounds like the value would have been `2`.

Reply:
If I have two trn.32xlarge work nodes, the value will be "2".

> Why it is 2 instead of 64 processes for 2 nodes w/ 32 cores each

We use mpi to execute processes `torchrun --nproc_per_node 32 --nnodes 2 --node_rank 0 --master_addr bert-mpi-training-worker-0.bert-mpi-training.default.svc.cluster.local --master_port 12355 e2e-test/bert-train.py` each `torchrun` will create "32" processes.

Reply:
Gotcha, forgot about that extra layer with `torchrun`.